modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Select/Select.tsx

```diff
@@ -241,7 +241,7 @@ const Main = ({
 	if (!multiple) {
 		if (
 			normalizedOptions.length &&
-			newValue?.[0] &&
+			!newValue?.[0] &&
```

Comment:
Hey @DiegoHu97 This change will generate a bug in the forms related to the predefined value, in short, the predefined value defined in the sidebar will not be reflected in the form builder. I would like to know what is the need to make this change since I tested without it and it seems that the undo/redo feature works, so do you really need to do this? :thinking: 


https://github.com/user-attachments/assets/a96acda4-b745-4b10-ac03-8617f1050bca

Reply:
Now I understand, thanks for clarifying @DiegoHu97. Could you provide a new solution that does what you want and doesn't cause this bug I mentioned above? Unfortunately, we cannot accept a PR that introduces a regression blocker for us. :smiling_face_with_tear:

Reply:
Sure @danvb112, we can find another alternative for that but I will need a bit of help. Could you give me the steps where the Selec From List fields is used in the portal to make sure my fix works? Apart from Structures, I don't know much about this field is used 🥲. 
Besides, about the Date picker change what do you think? If that change is fine then maybe we can move forward with it first. 
Thanks!

Reply:
Of course, @DiegoHu97  I can help you, I will try to find some time today to think of a solution that can solve this problem. And regarding the DatePicker changes, it looks good to me, I'll run the test suites on this PR to see if it's not breaking anything related to DatePicker, but if everything is ok with the tests, you can create another PR with DatePicker and leave this one with the Select from list, what do you think?

Reply:
@danvb112 Perfect then! If the tests passes, i can update tis PR to include only the Date Picker. This week I'm attending the bootcamp, so I won't have time to look into the Select field, if you don't have time neither, just tell me the usecases of the Select field and I can look into it :stuck\_out\_tongue\_closed\_eyes: . Thanks a lot!

Reply:
Hi @DiegoHu97, I spent some time trying to understand the problem, and I managed to find a solution that I believe will solve both problems, We actually need to refactor this component later since we are dealing with the logic of two components Single and Multiple selects in the same place, this can make it difficult things, we will do later. But for your problem, we need to add an extra validation in this condition to see if the newValue is an array and if it is empty, this is the case that will cover your problem, and we can keep the old logic avoiding causing the other bug I mentioned. I would extract this conditional into a function, to make things easier to understand:

```ts
	const hasUndefinedNewValue = () => {
		const hasNormalizedOptions = !!normalizedOptions.length;

		const hasNewValue = newValue?.[0] !== undefined;
		
		const isNewValueInvalid =
			hasNewValue &&
			!normalizedOptions.some((option) => option.value === newValue?.[0]);

		const isNewValueEmptyArray =
			Array.isArray(newValue) && !newValue.length;

		return (
			(hasNormalizedOptions && isNewValueInvalid) || isNewValueEmptyArray
		);
	};

	if (!multiple) {
		if (hasUndefinedNewValue()) {
			newValue = undefined;
		}
		
		...
```

Reply:
Could you please implement and test this solution @DiegoHu97?

Reply:
@danvb112 yes of course! thank you very much to look into it, I will close the PR until I implement the solution. Could you please give me the steps to test that the other case works well? The case of forms? thanks in advance!